cpp/ql/src/semmle/code/cpp/Variable.qll

```diff
@@ -176,7 +189,21 @@ class Variable extends Declaration, @variable {
 }
 
 /**
- * A particular declaration or definition of a C/C++ variable.
+ * A particular declaration or definition of a C/C++ variable. For example in
+ * the following code there are five variable declaration entries - two for
```

Comment:
I think these are just declarations, not definitions.

Reply:
There is a lot of scope for confusion between `Declaration`s and `DeclarationEntry`s which I don't think we do a great job of disentangling.  We actually have three types of things:
 - a `DeclarationEntry` for which `DeclarationEntry.isDefinition()` does not hold, such as `extern int a;`
 - a `DeclarationEntry` for which `DeclarationEntry.isDefinition()` does hold, such as `int a;`.  We call these things definitions.
 - a `Declaration`, such as the variable `a`.  These are easily confused with definitions as they are typically given the same location, but they are really a locationless conception of the variable / function / type.

I'd like to disambiguate the terminology but I think it should be in a separate PR, as it will affect other files.

Reply:
That's a good point.  I guess I had the C/C++ notion of "declaration" on my mind.

Reply:
s/For example/For Example,/

Reply:
Done, though I'm interested to hear why you prefer to have the comma?

Reply:
Re: declarations and definitions I attempted to clarify things a bit more in the qldoc comment for our `Declaration` class in https://github.com/Semmle/ql/pull/1925.  I think the name of our `Declaration` class itself is the real source of confusion (at least for new QL developers), but it's core to a lot of our and presumably customer QL code, so I'm not keen to change it.

If you know of anywhere else where we could be clearer, I'm happy to make further changes.  And we should discuss this more if we notice users getting significantly confused by this.

Reply:
No, there are no (further) changes that come to mind.  Would you like to make any more edits?  If not, I'll merge it in.

Reply:
I need to address your other comments, then that PR can be merged, yes.